src/Raven.Server/Documents/PeriodicBackup/BackupUploader.cs

```diff
@@ -0,0 +1,328 @@
+                client.PutBlob(key, stream, new Dictionary<string, string>
+                {
+                    { "Description", GetArchiveDescription() }
+                });
+
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"{ReportSuccess(AzureName)} container: {settings.StorageContainer}, with key: {key}");
+
+                var runner = new AzureRetentionPolicyRunner(_retentionPolicyParameters, client);
+                runner.Execute();
+            }
+        }
+
+        private void UploadToGoogleCloud(GoogleCloudSettings settings, Stream stream, Progress progress)
+        {
+            using (var client = new RavenGoogleCloudClient(settings, progress, TaskCancelToken.Token))
+            {
+                var key = CombinePathAndKey(settings.RemoteFolderName);
+                client.UploadObject(key, stream, new Dictionary<string, string>
+                {
+                    { "Description", GetArchiveDescription() }
+                });
+
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"{ReportSuccess(GoogleCloudName)} storage bucket: {settings.BucketName}");
+
+                var runner = new GoogleCloudRetentionPolicyRunner(_retentionPolicyParameters, client);
+                runner.Execute();
+            }
+        }
+
+        private string CombinePathAndKey(string path)
+        {
+            var prefix = string.IsNullOrWhiteSpace(path) == false ? $"{path}/" : string.Empty;
+            return $"{prefix}{_backupUploaderSettings.FolderName}/{_backupUploaderSettings.FileName}";
+        }
+
+        private void CreateUploadTaskIfNeeded<S, T>(S settings, Action<S, FileStream, Progress> uploadToServer, T uploadStatus, string targetName)
+            where S : BackupSettings
+            where T : CloudUploadStatus
+        {
+            if (PeriodicBackupConfiguration.CanBackupUsing(settings) == false)
+                return;
+
+            Debug.Assert(uploadStatus != null);
+
+            var localUploadStatus = uploadStatus;
+            var thread = PoolOfThreads.GlobalRavenThreadPool.LongRunning(_ =>
+            {
+                try
+                {
+                    Thread.CurrentThread.Priority = ThreadPriority.BelowNormal;
+                    NativeMemory.EnsureRegistered();
+
+                    using (localUploadStatus.UpdateStats(_isFullBackup))
+                    using (var fileStream = File.OpenRead(_backupUploaderSettings.BackupPath))
+                    {
+                        var uploadProgress = localUploadStatus.UploadProgress;
+                        try
+                        {
+                            localUploadStatus.Skipped = false;
+                            uploadProgress.ChangeState(UploadState.PendingUpload);
+                            uploadProgress.SetTotal(fileStream.Length);
+
+                            AddInfo($"Starting {uploadStatus.GetType().AssemblyQualifiedName}");
```

Comment:
You can replace uploadStatus.GetType().AssemblyQualifiedName with targetName

Reply:
done